pandas/tests/io/formats/test_format.py

```diff
@@ -368,6 +369,29 @@ def test_repr_min_rows(self):
             assert ".." not in repr(df)
             assert ".." not in df._repr_html_()
 
+    @pytest.mark.parametrize(
+        "data, format_option, expected_values",
+        [
+            ({"A": [12345.6789]}, "{:12.3f}", "12345.679"),
+            ({"A": [None]}, "{:.3f}", "None"),
+            ({"A": [""]}, "{:.2f}", ""),
+            ({"A": [112345.6789]}, "{:6.3f}", "112345.679"),
+        ],
+    )
+    def test_repr_float_formatting_html_output(
+        self, data, format_option, expected_values
+    ):
+        if format_option is not None:
+            set_option("display.float_format", format_option.format)
```

Comment:
Can you use the `option_context` context manager here instead?

Reply:
sure, sorry about that, have a better idea of the workflow now.